src/frontend/src/handler/create_table.rs

```diff
@@ -1157,14 +1177,31 @@ async fn derive_schema_for_cdc_table(
             table.pk_names().clone(),
         ))
     } else {
-        let columns = bind_sql_columns(column_defs)?;
-        // For table created by `create table t (*)` the constraint is empty, we need to
-        // retrieve primary key names from original table catalog if available
-        let pk_names = if let Some(original_catalog) = original_catalog {
-            original_catalog
+        let mut columns = bind_sql_columns(column_defs)?;
+        let pk_names = if let Some(args) = schema_change_args {
+            // If new_version_columns is provided, we are in the process of auto schema change.
+            // update the default value column since the default value column is not set in the
+            // column sql definition.
```

Comment:
You can take a look of `get_new_table_definition_for_cdc_table`.
Because auto schema change doesn't submit the new `ColumnDef` to frontend, we submit the `ColumnCatalog` parsed from message. So after mutation the definition of original catalog, there is no `DEFAULT expr` in the modified definition.

Reply:
Then shall we consider filling the `DEFAULT expr` clause when inferring a `ColumnDef` from `ColumnCatalog`?

Reply:
- How to do it? I didn't find a simple way to convert a `ColumnCatalog` back to `ColumnDef`
- We already have column catalog on hand, so I think it is unnecessary to convert back to `ColumnDef` and parse to `ColumnCatalog` again.

Reply:
> * I didn't find a simple way to convert a `ColumnCatalog` back to `ColumnDef`

IIUC we're essentially converting `ColumnCatalog` to `ColumnDef` here, though only data type is taken into consideration thus incomplete.

https://github.com/risingwavelabs/risingwave/blob/c59b3ebd3ee4c1d23a14f72bec48c0b66cf0dddc/src/frontend/src/handler/alter_table_column.rs#L113-L114

> * I think it is unnecessary to convert back to `ColumnDef` and parse to `ColumnCatalog` again

Theoretically yes, but what will the response look like when user querying `SHOW CREATE TABLE cdc_table` now?

Reply:
> IIUC we're essentially converting ColumnCatalog to ColumnDef here, though only data type is taken into consideration thus incomplete.

Specially, I mean the `ColumnOption` part.

> Theoretically yes, but what will the response look like when user querying SHOW CREATE TABLE cdc_table now?

The `DEFAULT expr` won't show up in the sql. And when create a new table, we won't capture the default value definition from upstream, I think the output is consistent in both case.

Reply:
> And when create a new table, we won't capture the default value definition from upstream

Would this be somehow inconsistent and surprising to users? Note that default values are also used when performing DML.

Reply:
> > And when create a new table, we won't capture the default value definition from upstream
> 
> Would this be somehow inconsistent and surprising to users? Note that default values are also used when performing DML.

For the case of auto schema mapping (`create table t (*)`), we can improve the feature in future to get the default value from upstream.
For the table with column definitions manually specified by user, I think they should also specify the default column by hand in the DDL.

Reply:
> For the case of auto schema mapping (`create table t (*)`), we can improve the feature in future to get the default value from upstream.

Sure. Can you please track it in an issue?

Reply:
Done. https://github.com/risingwavelabs/risingwave/issues/18359